### PR TITLE
feat: add admin dashboard, upload books

### PR DESCRIPTION
**Pull request by sreetamdas**
Upload books to Supabase with Prisma + API routes

For #62

### CODE REVIEW

File: src/components/Admin/UploadBook.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+import { Formik, FormikHelpers } from "formik";
+// eslint-disable-next-line import/no-named-as-default
```
Comment (bikkimahato) [inline src/components/Admin/UploadBook.tsx:3]: Commented code should not be pushed.

File: src/components/Admin/UploadBook.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		if (error) {
+			toast.error(error.message);
+		}
+
+		if (data) {
+			toast.success(`Uploaded ${imageFile.name} successfully!`);
+			const fileURL = getSupabaseFileURL(data?.Key);
+			const formValues: BookEntryProperties = { ...values, cover: fileURL };
+			const res = (await axios.post("/api/admin/books/add", formValues)).data;
+			// eslint-disable-next-line no-console
```
Comment (bikkimahato) [inline src/components/Admin/UploadBook.tsx:42]: Commented code should not be pushed.

File: src/components/Admin/UploadBook.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		const { data, error } = await uploadFileToSupabase(imageFile, { path: "/site/keebs/" });
+		if (error) {
+			toast.error(error.message);
+		}
+
+		if (data) {
+			toast.success(`Uploaded ${imageFile.name} successfully!`);
+			const fileURL = getSupabaseFileURL(data?.Key);
+			const formValues: BookEntryProperties = { ...values, cover: fileURL };
+			const res = (await axios.post("/api/admin/books/add", formValues)).data;
```
Comment (bikkimahato) [inline src/components/Admin/UploadBook.tsx:41]: You can wrap it in try catch for error handling.

File: src/components/Books/index.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+export type BookEntryProperties = {
+	name: string;
+	cover: string;
+	status: string;
+	author: string;
+};
+
+const BooksList = ({ results }: Pick<QueryDatabaseResponse, "results">) => {
+	// console.log(results);
```
Comment (bikkimahato) [inline src/components/Books/index.tsx:17]: Commented code should not be pushed.

File: src/components/Books/index.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+			const fileObject = properties["Cover"].files[0];
+			if ("type" in fileObject && fileObject.type === "file") {
+				details.cover = fileObject.file.url;
+			}
+
+			// Process "Tags"
+			if (properties["Tags"]["type"] === "select") {
+				const tagObject = properties["Tags"];
+				if (tagObject["type"] === "select") {
+					// details.status = { name: tagObject.select?.name ?? "", color: tagObject.select?.color };
```
Comment (bikkimahato) [inline src/components/Books/index.tsx:44]: Commented code should not be pushed.

File: src/pages/api/admin/books/add.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { BookEntryProperties } from "@/components/Books";
+import { prismaClient } from "@/utils/prismaClient";
+
+// @ts-expect-error BigInt prototype
```
Comment (bikkimahato) [inline src/pages/api/admin/books/add.ts:6]: Commented code should not be pushed.

File: src/pages/books.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import { BOOKS_DATABASE_ID } from "@/config";
+import { Center } from "@/styles/layouts";
+import { Title } from "@/styles/typography";
+
+const BooksPage = ({ results }: InferGetStaticPropsType<typeof getStaticProps>) => {
+	return (
+		<Fragment>
+			<DocumentHead
+				title="Books"
+				// description="Mechanical keyboards and their components that I own"
```
Comment (bikkimahato) [inline src/pages/books.tsx:17]: Commented code should not be pushed.

**Comment by sreetamdas** [comment]
@bikkimahato Thanks for the review! 

This PR is still a draft, but also, a couple of the comments are actually used for disabling certain ESLint and TypeScript warnings, so I will not be removing them.

**Pull request merged**